pkg/components/csi-snapshot-controller.go

```diff
@@ -25,34 +31,15 @@ func startCSISnapshotController(ctx context.Context, cfg *config.Config, kubecon
 	}
 
 	var (
-		whSA     = []string{"components/csi-snapshot-controller/webhook_serviceaccount.yaml"}
-		whCfg    = []string{"components/csi-snapshot-controller/webhook_config.yaml"}
-		whDeploy = []string{"components/csi-snapshot-controller/webhook_deployment.yaml"}
-		whSvc    = []string{"components/csi-snapshot-controller/webhook_service.yaml"}
-		cr       = []string{"components/csi-snapshot-controller/clusterrole.yaml"}
-		crb      = []string{"components/csi-snapshot-controller/clusterrolebinding.yaml"}
-		sa       = []string{"components/csi-snapshot-controller/serviceaccount.yaml"}
-		deploy   = []string{"components/csi-snapshot-controller/csi_controller_deployment.yaml"}
+		cr     = []string{"components/csi-snapshot-controller/clusterrole.yaml"}
+		crb    = []string{"components/csi-snapshot-controller/clusterrolebinding.yaml"}
+		sa     = []string{"components/csi-snapshot-controller/serviceaccount.yaml"}
+		deploy = []string{"components/csi-snapshot-controller/csi_controller_deployment.yaml"}
 	)
 
-	// Deploy Webhook
-	//nolint:nestif
-	if csiComps.Has(config.CsiComponentSnapshotWebhook) {
-		klog.Infof("deploying CSI snapshot webhook")
-		if err := assets.ApplyServiceAccounts(ctx, whSA, kubeconfigPath); err != nil {
-			return fmt.Errorf("apply service account: %w", err)
-		}
-		if err := assets.ApplyDeployments(ctx, whDeploy, renderTemplate, renderParamsFromConfig(cfg, nil), kubeconfigPath); err != nil {
-			return fmt.Errorf("apply deployment: %w", err)
-		}
-		if err := assets.ApplyValidatingWebhookConfiguration(ctx, whCfg, kubeconfigPath); err != nil {
-			return fmt.Errorf("apply validationWebhookConfiguration: %w", err)
-		}
-		if err := assets.ApplyDeployments(ctx, deploy, renderTemplate, renderParamsFromConfig(cfg, nil), kubeconfigPath); err != nil {
-			return fmt.Errorf("apply deployments: %w", err)
-		}
-	} else {
-		klog.Warningf("CSI snapshot webhook is disabled")
+	if err := deleteCSIWebhook(ctx, kubeconfigPath); err != nil {
+		klog.Warningf("Failed to delete resources of CSI Webhook, possibly corrupt state: %v", err)
```

Comment:
I'm not sure we need to `speculate` on the reason of the error in the message?

Reply:
Sure, I just copied it from the storage.go where topolvm deletion is happening